src/ColorUtils.php

```diff
@@ -2,33 +2,134 @@
 
-class ColorUtils {
-    public static function RGBAtoHEX(int $red, int $blue, int $green, ?int $alpha = null) {
-        $hex = [dechex($red), dechex($blue), dechex($green)];
-
-        if (!is_null($alpha)) {
-            $hex[] = dechex($alpha);
-        }
+/**
+ * ColourUtility is a PHP class that provides a set of utility functions for working with colours.
+ * 
+ * @method static array hexToRgb(string $hex)
+ * @method static string rgbToHex(int $r, int $g, int $b, $withPrefix = true)
+ * @method static int getBrightness(string $colour)
+ * @method static bool isDark(string $colour)
+ * @method static string getContrastColour(string $colour)
+ * @method static string getComplementaryColour(string $colour)
+ */
+class ColourUtility {
+  
+  /**
+   * The function converts a hexadecimal colour code to its corresponding RGB values.
+   * 
+   * @param string hex The "hex" parameter is a string representing a hexadecimal colour code.
+   * 
+   * @return array an array containing the RGB values of a given hexadecimal colour code. The array
+   * contains three elements, representing the red, green, and blue values respectively.
+   */
+  public static function hexToRgb(string $hex):array {
+    $hex = str_replace("#", "", $hex);
+    if(strlen($hex) == 3) {
+      $r = hexdec(substr($hex,0,1).substr($hex,0,1));
+      $g = hexdec(substr($hex,1,1).substr($hex,1,1));
+      $b = hexdec(substr($hex,2,1).substr($hex,2,1));
+    } else {
+      $r = hexdec(substr($hex,0,2));
+      $g = hexdec(substr($hex,2,2));
+      $b = hexdec(substr($hex,4,2));
+    }
+    return array($r, $g, $b);
+  }
+  
+  
+  /**
+   * The function converts RGB colour values to hexadecimal format with an optional prefix.
+   * 
+   * @param int r The red value of the RGB colour (an integer between 0 and 255).
+   * @param int g The green value of the RGB colour (an integer between 0 and 255).
+   * @param int b TThe blue value of the RGB colour (an integer between 0 and 255).
+   * @param withPrefix A boolean parameter that determines whether the returned hex string should include
+   * the "#" prefix or not. The default value is `true`.
+   * 
+   * @return string a string representing the hexadecimal value of the RGB colour passed as parameters.
+   */
+  public static function rgbToHex(int $r, int $g, int $b, $withPrefix = true):string {
+    $r = dechex($r);
+    $g = dechex($g);
+    $b = dechex($b);
 
-        foreach ($hex as $key => $value) {
-            if (strlen($value) !== 2) {
-                $hex[$key] = "0$value";
-            }
-        }
+    $r = strlen($r) == 1 ? "0" . $r : $r;
+    $g = strlen($g) == 1 ? "0" . $g : $g;
+    $b = strlen($b) == 1 ? "0" . $b : $b;
 
-        $hex = implode($hex);
+    $hex = $r . $g . $b;
 
-        return $hex;
+    if ($withPrefix) {
+      $hex = "#" . $hex;
     }
 
-    public static function HEXtoRGBA(string $hex): array
-    {
-        $rgba = str_split(str_replace("#", "", $hex), 2);
+    return $hex;
```

Comment:
```suggestion
    return $withPrex ? "#{$hex}" : $hex;
```
You had some pretty solid ternary work before, not sure what happened.
(you'll still need to update the rest of the function)

Reply:
> You had some pretty solid ternary work before, not sure what happened. (you'll still need to update the rest of the function)

I don't understand the problem?

Reply:
you're doing if ($withPrefix) { // prepend # to $hex } then returning it when you can just do a simple ternary at the bottom of the function